spec/tasks/import_films_from_csv_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rake'
+
+RSpec.describe 'rake import_films_from_csv:run', type: :task do
+  subject { Rake::Task["import_films_from_csv:run"].invoke(file_path) }
+
+  let(:file_path) { 'spec/fixtures/files/flix.csv' }
+
+  let(:file_content) { file_fixture('flix.csv').read }
+  let(:task) { Rake::Task["import_films_from_csv:run"] }
+
+  it 'triggers Film.import from csv' do
+    expect(Film).to receive(:import).with(
+      a_hash_including(title: 'Dick Johnson Is Dead')
+    )
+    
+    subject
```

Comment:
I think it's reasonable to use this London style of testing (i.e. "test X receives Y"). Since you are including the `rails_helper` (which spins up the whole app anyway), you might have considered the Chicago style – checking that it actually creates the object in the DB.

I wonder if this test might also include considerations for edge cases – e.g. where a required field is missing. However, it tests the "happy path" well.

Lastly, I wonder if this might have been better expressed as a unit test for the object you created. But that's really a matter of personal preference.

Reply:
Thanks! I have added the unit test for it and also the spec counting Film record [here](https://github.com/selvipeb/smartflix/pull/11/commits/d7655077ccaf867e768220600547076167de69a3)